packages/launcher/src/hooks/use-effect-router.ts

```diff
@@ -0,0 +1,48 @@
+import { useCallback, useMemo } from 'react';
+import { useSearchParams, useLocation } from 'react-router-dom';
+import useRouter from './use-router';
+
+export type SearchProps = Partial<Record<string, unknown>>;
+
+export type UseEffectRouterReturns = {
+    redirect(search?: SearchProps): void;
+    replace(search?: SearchProps): void;
+};
+
+const useEffectRouter = (): UseEffectRouterReturns => {
+    const { pathname, state } = useLocation();
+    const router = useRouter();
+    const params = useSearchParams();
+
+    const redirect = useCallback(
+        search => {
+            router.redirect(pathname, {
+                state,
+                ...params,
```

Comment:
this is state,no search

Reply:
> this is state,no search

u can look useRouter hook

Reply:
> this is state,no search

I think it should diffrent with state

Reply:
fuck,,,, I should change useRouter's behavior ..

it should handle this sence. wait me to change it

Reply:
> fuck,,,, I should change useRouter's behavior ..
> 
> it should handle this sence. wait me to change it

@myNameIsDu

Reply:
new redirect as 

```ts

    const redirect = useCallback(
        (path, params = {}) => {
            const { state, ...args } = params;

            const newPath = formatUrl(path, args);

            return navigate?.(newPath, {
                state,
            });
        },
        [navigate],
    );
```

Reply:
yeah,but this is give reactRouter state
<img width="493" alt="image" src="https://user-images.githubusercontent.com/53305259/157064805-2cf5d909-c922-4253-b99c-6fdc5d194e69.png">
<img width="670" alt="image" src="https://user-images.githubusercontent.com/53305259/157065158-3af81f74-ec35-431b-bdf1-64e43bc34a16.png">

Reply:
> yeah,but this is give reactRouter state <img alt="image" width="493" src="https://user-images.githubusercontent.com/53305259/157064805-2cf5d909-c922-4253-b99c-6fdc5d194e69.png"> <img alt="image" width="670" src="https://user-images.githubusercontent.com/53305259/157065158-3af81f74-ec35-431b-bdf1-64e43bc34a16.png">

wait me to change useRouter's behavior